python/tvm/tir/schedule/schedule.py

```diff
@@ -1733,6 +1733,258 @@ def after_set_scope(
+                A: T.Buffer[(128, 128), "float32"],
+                B: T.Buffer[(128, 128), "float32"],
+                C: T.Buffer[(128, 128), "float32"],
+            ) -> None:
+                # body
+                # with T.block("root")
+                for i_0, j_0 in T.grid(8, 8):
+                    for i_1_init, j_1_init in T.grid(16, 16):
+                        with T.block("init"):
+                            vi = T.axis.spatial(128, i_0 * 16 + i_1_init)
+                            vj = T.axis.spatial(128, j_0 * 16 + j_1_init)
+                            T.reads()
+                            T.writes(C[vi, vj])
+                            C[vi, vj] = T.float32(0)
+                    for k_0 in T.serial(8):
+                        with T.block("blockized_update"):
+                            vio, vjo, vko = T.axis.remap("SSR", [i_0, j_0, k_0])
+                            T.reads(
+                                C[vio * 16 : vio * 16 + 16, vjo * 16 : vjo * 16 + 16],
+                                A[vio * 16 : vio * 16 + 16, vko * 16 : vko * 16 + 16],
+                                B[vjo * 16 : vjo * 16 + 16, vko * 16 : vko * 16 + 16],
+                            )
+                            T.writes(C[vio * 16 : vio * 16 + 16, vjo * 16 : vjo * 16 + 16])
+                            A_1 = T.match_buffer(
+                                A[vio * 16 : vio * 16 + 16, vko * 16 : vko * 16 + 16],
+                                [16, 16],
+                                dtype="float32",
+                                offset_factor=1,
+                            )
+                            B_1 = T.match_buffer(
+                                B[vjo * 16 : vjo * 16 + 16, vko * 16 : vko * 16 + 16],
+                                [16, 16],
+                                dtype="float32",
+                                offset_factor=1,
+                            )
+                            C_1 = T.match_buffer(
+                                C[vio * 16 : vio * 16 + 16, vjo * 16 : vjo * 16 + 16],
+                                [16, 16],
+                                dtype="float32",
+                                offset_factor=1,
+                            )
+                            T.evaluate(
+                                T.tvm_mma_sync(
+                                    C_1.data,
+                                    C_1.elem_offset // 256,
+                                    A_1.data,
+                                    A_1.elem_offset // 256,
+                                    B_1.data,
+                                    B_1.elem_offset // 256,
+                                    C_1.data,
+                                    C_1.elem_offset // 256,
+                                    dtype="handle",
+                                )
+                            )
+
+        """
+        if isinstance(tensor_intrin, str):
+            tensor_intrin = String(tensor_intrin)
```

Comment:
Do we need this conversion? I suppose `PackedFuncValueConverter` is able to handle this :-)

Reply:
oh i see. It's because our C++ registry are trying to decide between String and TensorIntrin - then it makes sense to me